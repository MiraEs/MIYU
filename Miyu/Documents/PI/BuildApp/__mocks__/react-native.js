/* eslint-disable react/no-multi-comp */
import {
	createElement,
	Component,
} from 'react';
import PropTypes from 'prop-types';
const ReactNative = jest.genMockFromModule('react-native');

ReactNative.StyleSheet = {
	create: function create(styles) {
		return styles;
	},
	flatten: function flatten(...styles) {
		return Object.assign({}, ...styles);
	},
};

ReactNative.PixelRatio = {
	get() {
		return 1;
	},
};

ReactNative.Alert = {
	alert: jest.fn(),
};

ReactNative.StatusBar = {
	setBarStyle: jest.fn(),
	setHidden: jest.fn(),
};

ReactNative.Linking = {
	openURL: jest.fn(),
};

ReactNative.NativeModules = {
	PhotoUploader: {},
	SafariViewManager: {},
	UIManager: {
		measure: jest.fn(),
	},
	MarketingCloud: {
		trackAction: jest.fn(),
		trackState: jest.fn(),
	},
	LocalyticsManager: {
		tagScreen: jest.fn(),
		tagEvent: jest.fn(),
		tagCustomerLoggedOut: jest.fn(),
	},
	Instabug: {
		log: jest.fn(),
	},
	BNDevice: {},
	RNApplePayManager: {
		canMakePayments: jest.fn(() => Promise.resolve(true)),
		canMakePaymentsUsingNetworks: jest.fn(() => Promise.resolve(true)),
		paymentRequest: jest.fn(),
		paymentRequestDidFinish: jest.fn(),
		updateShippingMethods: jest.fn(),
		selectShippingMethod: jest.fn(),
		authorizedPayment: jest.fn(),
	},
};

ReactNative.requireNativeComponent = jest.fn();

ReactNative.NetInfo = {
	isConnected: {
		fetch: jest.fn(() => ({ done: jest.fn() })),
		addEventListener: jest.fn(),
		removeEventListener: jest.fn(),
	},
};

ReactNative.AppState = {
	addEventListener: jest.fn(),
};

ReactNative.CameraRoll = {
	getPhotos: jest.fn(() => Promise.resolve([])),
};

ReactNative.PushNotificationIOS = {
	setApplicationIconBadgeNumber: jest.fn(),
};

ReactNative.Keyboard = {
	dismiss: jest.fn(),
};

ReactNative.findNodeHandle = jest.fn();

ReactNative.ViewPropTypes = {
	style: jest.fn(),
};

ReactNative.NativeEventEmitter = class NativeEventEmitter {
	addListener = jest.fn(() => ({ remove: jest.fn() }));
};

class FlatList extends Component {
	render() {
		return createElement('FlatList', this.props);
	}
}

class SectionList extends Component {
	render() {
		return createElement('SectionList', this.props);
	}
}

class View extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('View', this.props, this.props.children);
	}
}

class ScrollView extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('ScrollView', this.props, this.props.children);
	}
}

class Switch extends Component {
	render() {
		return createElement('Switch');
	}
}

class Text extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Text', this.props, this.props.children);
	}
}

class TextInput extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TextInput', this.props, this.props.children);
	}
}

class TouchableOpacity extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableOpacity', this.props, this.props.children);
	}
}

class TouchableWithoutFeedback extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableWithoutFeedback', this.props, this.props.children);
	}
}

class ToolbarAndroid extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ToolbarAndroid', this.props, this.props.children);
	}
}

class Image extends Component {
	static propTypes = {
		children: PropTypes.any,
		style: () => {},
	};
	render() {
		return createElement('Image', this.props, this.props.children);
	}
}

class Icon extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('Icon', this.props, this.props.children);
	}
}

class ActivityIndicator extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('ActivityIndicator', this.props, this.props.children);
	}
}

ReactNative.Platform = {
	OS: 'ios',
	select: jest.fn((options) => options.ios),
};

ReactNative.InteractionManager = {
	runAfterInteractions: jest.fn(),
};

ReactNative.ActionSheetIOS = {
	showActionSheetWithOptions: jest.fn(),
};

class ListView extends Component {
	DataSource() {
		return {
			cloneWithRows: jest.fn(),
			cloneWithRowsAndSections: jest.fn(),
		};
	}
}

class AppRegistry {
	registerComponent() {}
}

class TouchableHighlight extends Component {
	static propTypes = {
		children: PropTypes.any,
	};
	render() {
		return createElement('TouchableHighlight', this.props, this.props.children);
	}
}
class BackHandler extends Component {
	static addEventListener() {}
	static exitApp() {}
	static removeEventListener() {}
}

class TouchableNativeFeedback extends Component {
	static Ripple() {
	}
}

class Navigator extends Component {
	render() {
	}
}

class RefreshControl extends Component {
	render() {
		return createElement('RefreshControl', this.props);
	}
}

class LayoutAnimation extends Component {
	static configureNext() {
	};

	static Presets() {
		return {
			easeInEaseOut: jest.fn(),
		};
	};
}

class Dimensions {
	static get() {
		return 0;
	}
}

class Animated extends Component {

	static Value() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static ValueXY() {
		return {
			getLayout: () => {},
			interpolate: () => {},
			setOffset: () => {},
			setValue: () => {},
		};
	}

	static event() {}
}

class AnimatedView extends Component {
	render() {
		return createElement('AnimatedView');
	}
}

class AnimatedImage extends Component {

}

class AnimatedText extends Component {

}

Animated.View = AnimatedView;

Animated.Image = AnimatedImage;

Animated.Text = AnimatedText;

class PanResponder {
	static create() {
		return {panHandlers: {}};
	}
}

global.__fbBatchedBridgeConfig = {
	remoteModuleConfig: [],
	localModulesConfig: [],
};

ReactNative.View = View;
ReactNative.ScrollView = ScrollView;
ReactNative.FlatList = FlatList;
ReactNative.ListView = ListView;
ReactNative.Text = Text;
ReactNative.TextInput = TextInput;
ReactNative.TouchableOpacity = TouchableOpacity;
ReactNative.TouchableHighlight = TouchableHighlight;
ReactNative.TouchableWithoutFeedback = TouchableWithoutFeedback;
ReactNative.ToolbarAndroid = ToolbarAndroid;
ReactNative.Image = Image;
ReactNative.AppRegistry = AppRegistry;
ReactNative.Icon = Icon;
ReactNative.ActivityIndicator = ActivityIndicator;
ReactNative.BackHandler = BackHandler;
ReactNative.TouchableNativeFeedback = TouchableNativeFeedback;
ReactNative.Navigator = Navigator;
ReactNative.RefreshControl = RefreshControl;
ReactNative.LayoutAnimation = LayoutAnimation;
ReactNative.Dimensions = Dimensions;
ReactNative.Animated = Animated;
ReactNative.PanResponder = PanResponder;
ReactNative.SectionList = SectionList;
ReactNative.Switch = Switch;

module.exports = ReactNative;
