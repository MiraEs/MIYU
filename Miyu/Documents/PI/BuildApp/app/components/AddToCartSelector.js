import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	LayoutAnimation,
	StyleSheet,
	View,
} from 'react-native';
import {
	NO_THANK_YOU,
	I_DONT_NEED_THIS,
	CHECK_BOX_OPTION,
} from '../constants/productDetailConstants';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withNavigation } from '@expo/ex-navigation';
import {
	addSessionCartItems,
	updateCartItemBounce,
} from '../actions/CartActions';
import { getProductComposite } from '../actions/ProductDetailActions';
import productConfigurationsActions from '../actions/ProductConfigurationsActions';
import styles from '../lib/styles';
import AddToCartButton from '../components/AddToCartButton';
import tracking from '../lib/analytics/tracking';
import helpers from '../lib/helpers';
import { trackAction } from '../actions/AnalyticsActions';
import { QuantitySelector } from 'BuildLibrary';

const componentStyles = StyleSheet.create({
	quantitySelectorShown: {
		width: 132,
	},
	quantitySelectorHidden: {
		width: 44,
		marginRight: 1,
	},
});

@withNavigation
export class AddToCartSelector extends Component {

	constructor(props) {
		super(props);
		this.state = {
			addToCartEnabled: true,
			loading: false,
			quantitySelectorMode: false,
			quantitySelectorVisible: true,
		};

		this.addToCartButton = {
			reset: helpers.noop,
			queueChange: helpers.noop,
		};
	}

	onAddToCartPress = () => {
		this.addToCartButton.queueChange({ text: 'Adding...' });
		const { productConfigurationId, actions, productComposite, compositeId, uniqueId } = this.props;
		if (!this.props.hideQuantitySelector) {
			LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
		}
		this.setState({
			addToCartEnabled: false,
			quantitySelectorVisible: false,
		}, () => {
			if (!productComposite) {
				actions.createProductConfiguration({ compositeId, uniqueId, productConfigurationId })
					.then(this.validate)
					.catch(helpers.noop)
					.done();
			} else {
				this.validate();
			}
		});
	};

	isOptional = (pricedOption) => {
		return (pricedOption.value.toUpperCase() === NO_THANK_YOU || pricedOption.value.toUpperCase() === I_DONT_NEED_THIS);
	};

	isOptionalPricedOptionGroup = (pricedOptionGroup) => {
		return pricedOptionGroup.pricedOptions.filter((pricedOption) =>
			this.isOptional(pricedOption) || pricedOption.inputType.toUpperCase() === CHECK_BOX_OPTION).length > 0;
	};

	validate = () => {
		const {
			productComposite,
			onAvailableByLocation,
			onSquareFootageBased,
			onHasOptionGroups,
			validateAvailability,
			selectedPricedOptions,
		} = this.props;

		if (productComposite.availableByLocation && onAvailableByLocation) {
			onAvailableByLocation();
			return this.resetAddToCartButton();
		}

		if (validateAvailability && !validateAvailability()) {
			return this.resetAddToCartButton();
		}

		if (productComposite.squareFootageBased && this.props.quantity === 0) {
			onSquareFootageBased();
			return this.resetAddToCartButton();
		}
		if (productComposite.pricedOptionGroups.length) {
			const { pricedOptionGroups } = productComposite;
			const requiredPricedOptionGroups = pricedOptionGroups
			.filter((optionGroup) => !this.isOptionalPricedOptionGroup(optionGroup));
			const optionalPricedOptionGroupNames = pricedOptionGroups
			.filter((optionGroup) => this.isOptionalPricedOptionGroup(optionGroup))
			.map((optionGroup) => optionGroup.optionName);
			const requiredSelectedOptions = selectedPricedOptions.filter((option) => option.pricedOptions.length)
			.map((option) => option.optionName)
			.filter((optionName) => !optionalPricedOptionGroupNames.includes(optionName));
			if (requiredSelectedOptions.length === requiredPricedOptionGroups.length) {
				return this.addItemsToCart();
			}
			onHasOptionGroups(requiredSelectedOptions);
			return this.resetAddToCartButton();
		}
		return this.addItemsToCart();
	};

	getSelectorStyle = () => {
		if (this.state.quantitySelectorMode) {
			return componentStyles.quantitySelectorShown;
		}
		return componentStyles.quantitySelectorHidden;
	};

	addItemsToCart = () => {
		const { actions, cart, productComposite, uniqueId, selectedPricedOptions } = this.props,
			{ sessionCartId } = cart,
			{ availability } = productComposite,
			sessionCartItems = [],
			pricedOptions = [];
		selectedPricedOptions.map((optionGroup) => {
			if (optionGroup) {
				optionGroup.pricedOptions.map((option) => {
					pricedOptions.push({
						keyCode: option.keyCode,
						pricedOptionId: option.pricedOptionId,
					});
				});
			}
		});
		tracking.trackNewCartCreated(cart, () => {
			actions.updateCartItemBounce(true);
		});
		this.trackAddToCart();

		sessionCartItems.push({
			productUniqueId: uniqueId,
			quantity: this.props.quantity,
			pricedOptions,
		});
		actions.addSessionCartItems({
			postalCode: availability.zipCode,
			postalCodeSource: availability.isAvailableInZip ? 'confirmed' : null,
			sessionCartItems,
			sessionCartId,
		}).then((results) => {
			const parentKey = results.cartItems[0].itemKey;
			this.addToCartButton.queueChange({
				text: 'Added To Cart',
				itemAdded: true,
			}, () => {
				this.handleUpsell(parentKey);
				this.resetAddToCartButton();
			});
		}).catch(() => {
			this.addToCartButton.queueChange({ text: 'Unable To Add' }, this.resetAddToCartButton);
		}).done();
	};

	handleUpsell = (parentKey) => {
		const { compositeId, selectedFinish, productComposite, productConfigurationId } = this.props;
		const { recommendedOptions, accessories } = productComposite;
		if (recommendedOptions && recommendedOptions.length || accessories && accessories.length) {
			this.props.navigation.getNavigator('root').push('productUpsell', {
				addedProductQuantity: this.props.quantity,
				finish: selectedFinish,
				recommendedOptions,
				compositeId,
				parentKey,
				accessories,
				productConfigurationId,
			});
		}
	};

	resetAddToCartButton = () => {
		this.setState({ quantity: 1 });
		this.addToCartButton.reset(() => {
			if (!this.props.hideQuantitySelector) {
				LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
			}
			this.setState({
				quantitySelectorMode: false,
				quantitySelectorVisible: true,
				addToCartEnabled: true,
			});
		});
	};

	onToggleSelectors = (quantitySelectorMode) => {
		LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
		this.setState({ quantitySelectorMode });
	};

	getQuantitySelector = () => {
		if (this.state.quantitySelectorVisible && !this.props.hideQuantitySelector) {
			return (
				<QuantitySelector
					style={this.getSelectorStyle()}
					quantity={this.props.quantity}
					onUpdateQuantity={this.props.onUpdateQuantity}
					disableDelete={true}
					allowZero={false}
					onToggleSelectors={this.onToggleSelectors}
					theme="primary"
					onInputFocus={this.props.onInputFocus}
				/>
			);
		}
	};

	updateQuantity = (quantity) => {
		this.setState({ quantity });
	};

	trackAddToCart = () => {
		const { selectedFinish, productComposite: { manufacturer, productId, rootCategory }, quantity } = this.props;
		const { finish, sku, uniqueId, pricebookCostView: { cost } } = selectedFinish;
		const contextData = {
			manufacturer,
			productId,
			finish,
			sku,
			uniqueId,
			cost,
			quantity,
			categoryId: rootCategory.categoryId,
			categoryName: rootCategory.categoryName,
		};
		this.props.actions.trackAction(this.props.trackAction, contextData);
	};

	render() {
		return (
			<View style={[styles.elements.centeredFlexRow, styles.elements.flex]}>
				{this.getQuantitySelector()}
				<AddToCartButton
					ref={(ref) => {
						if (ref) {
							this.addToCartButton = ref;
						}
					}}
					style={styles.elements.flex}
					text="Add To Cart"
					disabled={!this.state.addToCartEnabled}
					onPress={this.onAddToCartPress}
				/>
			</View>
		);
	}

}

AddToCartSelector.propTypes = {
	actions: PropTypes.object,
	cart: PropTypes.object,
	compositeId: PropTypes.number,
	hideQuantitySelector: PropTypes.bool,
	onAvailableByLocation: PropTypes.func,
	onHasOptionGroups: PropTypes.func.isRequired,
	onSquareFootageBased: PropTypes.func.isRequired,
	onUpdateQuantity: PropTypes.func,
	productConfigurationId: PropTypes.string.isRequired,
	productComposite: PropTypes.object,
	quantity: PropTypes.number.isRequired,
	uniqueId: PropTypes.number,
	selectedFinish: PropTypes.object,
	trackAction: PropTypes.string,
	validateAvailability: PropTypes.func,
	onInputFocus: PropTypes.func,
	selectedPricedOptions: PropTypes.array,
	navigation: PropTypes.shape({
		getNavigator: PropTypes.func,
	}),
};

const mapStateToProps = function (state, ownProps) {
	const configuration = state.productConfigurationsReducer[ownProps.productConfigurationId];
	if (!configuration) {
		return {
			cart: state.cartReducer.cart,
			uniqueId: ownProps.selectedFinish && ownProps.selectedFinish.uniqueId,
		};
	}
	const {
		compositeId,
		uniqueId,
		selectedFinish,
		selectedPricedOptions,
	} = configuration;
	return {
		cart: state.cartReducer.cart,
		compositeId,
		productComposite: state.productsReducer[compositeId],
		selectedFinish,
		selectedPricedOptions,
		uniqueId,
	};
};

const mapDispatchToProps = function (dispatch) {
	return {
		actions: bindActionCreators({
			addSessionCartItems,
			createProductConfiguration: productConfigurationsActions.createProductConfiguration,
			getProductComposite,
			trackAction,
			updateCartItemBounce,
		}, dispatch),
	};
};

export default connect(
	mapStateToProps,
	mapDispatchToProps,
)(AddToCartSelector);
