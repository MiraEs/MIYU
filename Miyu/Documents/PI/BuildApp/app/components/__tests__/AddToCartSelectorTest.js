
jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('BuildLibrary');
jest.mock('BuildNative');
jest.mock('../../../app/components/Form', () => 'Form');
jest.mock('../../../app/components/FormInput', () => 'FormInput');
jest.mock('../../../app/components/AddToCartButton', () => 'AddToCartButton');

jest.unmock('react-native');
import 'react-native';
import React from 'react';
import { AddToCartSelector } from '../AddToCartSelector';

const defaultProps = {
	compositeId: 1234,
	onHasOptionGroups: jest.fn(),
	onSquareFootageBased: jest.fn(),
	quantity: 22,
	uniqueId: 8767,
};

describe('AddToCartSelector component', () => {

	it('should render correctly with only required props', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with hidden quantity selector', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				hideQuantitySelector={true}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with onUpdateQuantity function', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				onUpdateQuantity={jest.fn()}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with productComposite prop', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				productComposite={{}}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with selectedFinish prop', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				selectedFinish={{}}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with validateAvailability prop', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				validateAvailability={jest.fn()}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render with onInputFocus prop', () => {
		const tree = require('react-test-renderer').create(
			<AddToCartSelector
				{...defaultProps}
				onInputFocus={jest.fn()}
			/>
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

});
