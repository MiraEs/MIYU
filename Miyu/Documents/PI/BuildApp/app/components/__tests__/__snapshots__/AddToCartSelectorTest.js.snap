// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartSelector component should render correctly with only required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={undefined}
    onToggleSelectors={[Function]}
    onUpdateQuantity={undefined}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with hidden quantity selector 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with onInputFocus prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={[Function]}
    onToggleSelectors={[Function]}
    onUpdateQuantity={undefined}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with onUpdateQuantity function 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={undefined}
    onToggleSelectors={[Function]}
    onUpdateQuantity={[Function]}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with productComposite prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={undefined}
    onToggleSelectors={[Function]}
    onUpdateQuantity={undefined}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with selectedFinish prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={undefined}
    onToggleSelectors={[Function]}
    onUpdateQuantity={undefined}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;

exports[`AddToCartSelector component should render with validateAvailability prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
      },
      Object {
        "flexGrow": 1,
      },
    ]
  }
>
  <QuantitySelector
    allowZero={false}
    disableDelete={true}
    onInputFocus={undefined}
    onToggleSelectors={[Function]}
    onUpdateQuantity={undefined}
    quantity={22}
    style={
      Object {
        "marginRight": 1,
        "width": 44,
      }
    }
    theme="primary"
  />
  <AddToCartButton
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "flexGrow": 1,
      }
    }
    text="Add To Cart"
  />
</View>
`;
